src/onegov/search/mixins.py

```diff
@@ -258,10 +258,12 @@ def create_tsvector_string(*cols):
 
         :param cols: column names to be indexed
         :return: tsvector string for multiple columns i.e. for columns title
-        and body: coalesce(title, '') || ' ' || coalesce(body, '')
+        and body: coalesce([title], '') || ' ' || coalesce([body], '')
+        Note that the '\"{}\"' escapes the column name in case it is a
+        keyword, see http://www.postgresql.org/docs/current/static/sql-keywords-appendix.html
         """
-        base = "coalesce({}, '')"
-        ext = " || ' ' || coalesce({}, '')"
```

Comment:
I don't think this will work, since you won't get strings as inputs, it happens to work for columns, but in general you are probably better off using `sqlalchemy.func.coalesce` and `sqlalchemy.func.concat` on the input. That will ensure you will generate valid SQL statement, regardless of whether the column is an actual column, or some arbitrarily complex expression (which is what you will get with `hybrid_property`).

Reply:
This also should do the necessary escaping on column names for you.

Reply:
Adjusted tsvector string generation to support hybrid_properties

Reply:
This is still incorrect, you shouldn't use strings at all, you should build the expression using the sqlalchemy functions, as long as you use string interpolation, it won't work correctly in every case.

So `base` would look something like `base = func.coalesce(expr, '')` and `ext` something like `func.concat(left_expr, func.concat(' ', func.coalesce(right_expr, ''))`

Reply:
Weird, I did exactly that, re-indexing would not work!